Problem compiling with clang on Windows

Testing with clang on arm64 on Windows we see:

```
** libs
using C compiler: 'clang version 16.0.5'
using C++ compiler: 'clang version 16.0.5'
clang++ -std=gnu++17  -I"C:/PROGRA~1/R/R-devel/include" -DNDEBUG -DSTRICT_R_HEADERS -I'C:/Users/jeroen/AppData/Local/R/win-library/4.4/Rcpp/include'   -I"/c/msys64/home/jeroen/aarch64-w64-mingw32.static.posix/include"     -O2 -Wall    -c RcppExports.cpp -o RcppExports.o
In file included from RcppExports.cpp:5:
In file included from C:/Users/jeroen/AppData/Local/R/win-library/4.4/Rcpp/include/Rcpp.h:38:
C:/Users/jeroen/AppData/Local/R/win-library/4.4/Rcpp/include/Rcpp/Environment.h:91:24: error: no matching function for call to 'R_lsInternal'
                return R_lsInternal( env, all ? TRUE : FALSE ) ;
                       ^~~~~~~~~~~~
C:/PROGRA~1/R/R-devel/include/Rinternals.h:533:6: note: candidate function not viable: no known conversion from 'int' to 'Rboolean' for 2nd argument
SEXP R_lsInternal(SEXP, Rboolean);
     ^
1 error generated.
make: *** [C:/PROGRA~1/R/R-devel/etc/Makeconf:289: RcppExports.o] Error 1
ERROR: compilation failed for package 'later'
* removing 'C:/Users/jeroen/AppData/Local/R/win-library/4.4/later'

```

Previous topics in https://github.com/RcppCore/Rcpp/issues/1148  and https://github.com/RcppCore/Rcpp/issues/897 suggest a likely issue is something is included that redefines `TRUE`.

--- a/inst/include/later.h
+++ b/inst/include/later.h
@@ -12,12 +12,6 @@
 #define STRICT_R_HEADERS
 #endif
 
-#include <Rinternals.h>
-
-// Needed for R_GetCCallable on R 3.3 and older; in more recent versions, this
-// is included via Rinternals.h.
-#include <R_ext/Rdynload.h>
-
 #ifdef _WIN32
 #define WIN32_LEAN_AND_MEAN
 // Taken from http://tolstoy.newcastle.edu.au/R/e2/devel/06/11/1242.html
@@ -29,6 +23,14 @@
 #else // _WIN32
 #include <pthread.h>
 #endif // _WIN32
+
+#include <Rinternals.h>
+
+// Needed for R_GetCCallable on R 3.3 and older; in more recent versions, this
+// is included via Rinternals.h.
+#include <R_ext/Rdynload.h>
+
+
 
 namespace later {
 

